lib/src/formatter.dart

```diff
@@ -17,13 +18,16 @@ class LcovFormatter implements Formatter {
   final Resolver resolver;
   LcovFormatter(this.resolver);
 
-  Future<String> format(Map hitmap) async {
+  Future<String> format(Map hitmap, {List<String> reportOn}) async {
     var buf = new StringBuffer();
-
+    var reportOnPaths = reportOn != null ? reportOn.map((path) => new File(path).absolute.path) : [];
     hitmap.forEach((key, v) {
       var source = resolver.resolve(key);
       if (source == null) {
         return;
+      } else if (reportOn != null && reportOn.length > 0 &&
+                 reportOnPaths.where((p) => source.startsWith(p)).length == 0) {
```

Comment:
Instead of `where` use `!reportsOnPaths.any((p) => ...)`

You calculate the whole length of the enumerable, which can be avoided.